shared/src/main/scala/com/raquo/domtypes/defs/attrs/MathMLDefs.scala

```diff
@@ -0,0 +1,22 @@
+package com.raquo.domtypes.defs.attrs
+
+import com.raquo.domtypes.common.MathMLTagType
+import com.raquo.domtypes.common.AttrDef
+
+object MathMlAttrDefs {
+
+  // @nikita: Wasn't this the part we concluded didn't work?
+  /**
+    * If you break open the browser tools and noodle around with mathml elements, you'll see that they have a bunch of attributes.,
+    * you may find they have little effect or are "undefined" in the browser.
+    *
+    * This would appear to be because they aren't part of the browser API?
+    *
+    * We take a pass on these here.
+    *
+    */
+val defs: List[AttrDef] = List(
+
+
+  )
+}
```

Comment:
No – there's a difference between markup attributes and DOM properties. The properties are what's missing – as evidenced by `undefined` in the browser, as we've seen. But the attributes of course still exist. I'm not proficient in mathml, but I assume that specifying them is in fact needed to accomplish some things.

Markup attributes are typically specified in the mathml markup, but they can also be set via JS, like this: `$0.setAttribute('displaystyle', 'normal')` (You can try that on [this page](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mstyle), if you inspect & select the `<mstyle>` element in their example – you will see the formula change its layout.

Looks like this is the list of attributes: https://developer.mozilla.org/en-US/docs/Web/MathML/Attribute

Don't add the ones with the trash icon.

With the attributes, the most important thing is to get their types /codecs right. See the types in the table above.
- If the type is not mentioned, it's just a strings, so for them the codec would be just StringAsIs.
- At the moment we encode enum-type values as strings as well – just specify their allowed values in the scaladoc.
- For booleans, [this page](https://developer.mozilla.org/en-US/docs/Web/MathML/Values#mathml-specific_types) says they they are encoded as "true" and "false" strings, so for that, we should use BooleanAsTrueFalseStringCodec
- If not sure about a particular attribute, just set the type to string. It's a safe fallback, because all those values are represented as strings in the mathml markup.

If it's not too much work, it would be nice to indicate which element(s) each of these applies to in the scaladoc (the info is in the table)

Reply:
I believe this could be done in a follow up PR? 

I need to read the text carefully, but want to try and get through the tags first.